getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.kotlin.demo
+
+import com.google.cloud.storage.Acl
+import com.google.cloud.storage.Storage
+import com.google.cloud.storage.StorageOptions
+import com.google.cloud.vision.v1.*
+import com.google.cloud.vision.v1.Feature.Type
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.awt.Polygon
+import java.io.FileOutputStream
+import java.io.PrintStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import javax.imageio.ImageIO
+
+val storage: Storage = StorageOptions.getDefaultInstance().service
+val bucket = storage.get("cloud-kotlin-samples")
+val bucketUri = "gs://cloud-kotlin-samples"
+
+const val RESOURCES_PATH = "resources/"
+const val EMOJIS_PATH = RESOURCES_PATH + "emojis/"
+
+enum class Emoji {
+    JOY, ANGER, SURPRISE, SORROW, NONE
+}
+
+val emojiPic = mapOf(
+    Emoji.JOY to "joy.png",
+    Emoji.ANGER to "anger.png",
+    Emoji.SURPRISE to "surprise.png",
+    Emoji.SORROW to "sorrow.png",
+    Emoji.NONE to "none.png"
+)
+
+//Returns best emoji based on detected emotions likelihoods
+fun bestEmoji(annotation: FaceAnnotation): Emoji {
+    val emotionsLikelihood: Array<Likelihood> = arrayOf(Likelihood.VERY_LIKELY, Likelihood.LIKELY, Likelihood.POSSIBLE)
+    val emotions = mapOf(
+        Emoji.JOY to annotation.joyLikelihood,
+        Emoji.ANGER to annotation.angerLikelihood,
+        Emoji.SURPRISE to annotation.surpriseLikelihood,
+        Emoji.SORROW to annotation.sorrowLikelihood
+    )
+
+    for (emotionLikelihood in emotionsLikelihood) {
+        for (emotion in emotions) {
```

Comment:
I think you can simplify this with a if `emotionsLikelihood contains emotion`

Reply:
I don't really understand. 

This is what I'm trying to do:
1. I'm more worried of the order: "Let's first find which of the 4 emotions (in this order: JOY, ANGER, SURPRISE, SORROW) has a VERY_LIKELY likelihood". If for instance JOY and SURPRISE are VERY_LIKELY, we will select JOY because of the order. We do that for each of the likelihoods, again in order.

2. I need to return the emotion that matches the likelihood so I think I still need the double loop.

Reply:
Good point!